src/components/ExportLicenseAsCSVModal/ExportLicenseAsCSVModal.js

```diff
@@ -0,0 +1,175 @@
+import css from './ExportLicenseAsCSVModal.css';
+
+export default class ExportLicenseAsCSVModal extends React.Component {
+  static propTypes = {
+    onCompareLicenseTerms: PropTypes.func,
+    onClose: PropTypes.func,
+    selectedLicenses: PropTypes.arrayOf(PropTypes.string),
+    terms: PropTypes.arrayOf(PropTypes.object),
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.licenseInformation = {
+      name: <FormattedMessage id="ui-licenses.prop.name" />,
+      startDate: <FormattedMessage id="ui-licenses.prop.startDate" />,
+      endDate: <FormattedMessage id="ui-licenses.prop.endDate" />,
+      status: <FormattedMessage id="ui-licenses.prop.status" />,
+      type: <FormattedMessage id="ui-licenses.prop.type" />,
+    };
+
+    this.terms = {
+      value: <FormattedMessage id="ui-licenses.prop.termValue" />,
+      note: <FormattedMessage id="ui-licenses.prop.internalNote" />,
+      publicNote: <FormattedMessage id="ui-licenses.prop.publicNote" />,
+      internal: <FormattedMessage id="ui-licenses.prop.termVisibility" />,
+    };
+
+    this.termNames = props.terms.reduce((map, term) => {
+      map[term.name] = term.label;
+      return map;
+    }, {});
+
+    this.state = {
+      licenseInformation: mapValues(this.licenseInformation, () => false),
+      terms: mapValues(this.terms, () => false),
+      termNames: mapValues(this.termNames, () => false)
+    };
+  }
+
+  renderCheckboxesList = (section) => {
+    return (
+      <>
+        <Checkbox
+          checked={Object.values(this.state[section]).includes(false) !== true}
+          label={<strong><FormattedMessage id={`ui-licenses.exportLicensesModal.${section}`} /></strong>}
+          onChange={(e) => this.toggleSelectSection(e, section)}
+          value={section}
+        />
+        <Layout className="padding-start-gutter">
+          {
+          Object.entries(this[section]).map(([prop, value], index) => (
+            <Checkbox
+              checked={this.state[section][prop]}
+              key={index}
+              label={value}
+              labelClass={css.labelClass}
+              name={prop}
+              onChange={(e) => this.updateSelection(e, section)}
+              value={prop}
+            />
+          ))
+        }
+        </Layout>
+      </>
+    );
+  }
+
+  renderExportLicenseAsCSVModal = () => {
+    const { licenseInformation, terms, termNames } = this.state;
+    const footer = (
+      <ModalFooter>
+        <Button
+          buttonStyle="primary"
+          disabled={Object.values({ ...licenseInformation, ...terms, ...termNames }).includes(true) !== true}
+          id="export-licenses-modal-save-button"
+          onClick={() => {
+            const payload = {
+              ids: this.props.selectedLicenses,
+              include: { ...pickBy(licenseInformation), ...{ 'customProperties': pickBy(termNames) } },
+              terms: pickBy(terms)
+            };
+
+            this.props.onCompareLicenseTerms(payload)
```

Comment:
How long might this take? Should we be putting up a Spinner or is relatively quick (1-3 seconds)?

Reply:
Will add it once I test this when the backend is ready